release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,321 @@
+                        List<ReleaseNoteText> relatedPrs = sameAsItems.get(topicPrNumber);
+                        if (relatedPrs == null) relatedPrs = new ArrayList<>();
+                        relatedPrs.add(releaseNoteText);
+                        sameAsItems.put(topicPrNumber, relatedPrs);
+                    }
+                }
+                categorizeReleaseNoteText(releaseNoteText);
+            }
+        }
+    }
+
+    void categorizeReleaseNoteText(ReleaseNoteText rnText) {
+        checkReleaseNoteCategory(rnText);
+        Arrays.stream(Category.values()).forEach(category -> {
+            if (rnText.category.equals(category)) {
+                List<ReleaseNoteText> releaseNoteTexts = categoryMap.get(category);
+                if (releaseNoteTexts == null) releaseNoteTexts = new ArrayList<>();
+                if (!isContainedInSameAsItems(rnText))
+                    releaseNoteTexts.add(rnText);
+                categoryMap.put(category, releaseNoteTexts);
+            }
+        });
+    }
+
+    void assortSameAsItems() {
+        for (Entry<String, List<ReleaseNoteText>> entry : sameAsItems.entrySet()) {
+            String topicPrNumber = entry.getKey();
+            List<ReleaseNoteText> releaseNoteTextsInSameAs = entry.getValue();
+
+            Arrays.stream(Category.values()).forEach(category -> {
+                List<ReleaseNoteText> releaseNoteTextsInACategory = categoryMap.get(category);
+                if (releaseNoteTextsInACategory != null) {
+                    releaseNoteTextsInACategory.forEach(rnInfo -> {
+                        if (rnInfo.prNumbers.get(0).equals(topicPrNumber)) {
+                            releaseNoteTextsInSameAs.forEach(from -> {
+                                merge(from, rnInfo);
+                            });
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    void merge(ReleaseNoteText from, ReleaseNoteText to) {
+        if (from.text != null && from.text.length() > 0) {
+            to.text = to.text + " " + from.text;
+        }
+        if (DEBUG != null) System.err.printf("merged RN text: %s\n", to.text);
+        to.prNumbers.addAll(from.prNumbers);
+    }
+
+    void outputReleaseNote() {
+        StringBuilder builder = new StringBuilder();
+        builder.append("## Summary\n\n");
+
+        Arrays.stream(Category.values()).forEach(category -> {
+            List<ReleaseNoteText> releaseNotes = categoryMap.get(category);
+            if (releaseNotes != null && !releaseNotes.isEmpty()) {
+                builder.append(String.format("## %s\n", category.getDisplayName()));
+                builder.append(getFormattedReleaseNotes(category, releaseNotes)).append("\n");
+            }
+        });
+        System.out.println(builder.toString());
+    }
+
+    String getFormattedReleaseNotes(Category category, List<ReleaseNoteText> releaseNotes) {
```

Comment:
The first argument `category` is never used.

Reply:
I removed it. Thank you.